File: spring-boot/src/main/java/com/artivisi/techstack/registration/domain/Registration.java
package com.artivisi.techstack.registration.domain;

import java.time.OffsetDateTime;
import java.time.ZoneOffset;
import java.time.format.DateTimeFormatter;
import java.util.UUID;

public record Registration(
        UUID id,
        String email,
        String fullName,
        String phone,
        OffsetDateTime createdAt
) {
    private static final DateTimeFormatter UTC_PATTERN =
            DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss");

    public String createdAtFormatted() {
        return createdAt.withOffsetSameInstant(ZoneOffset.UTC).format(UTC_PATTERN) + " UTC";
    }
}

File: golang/internal/domain/registration.go
package domain

import "time"

type Registration struct {
	ID        string
	Email     string
	FullName  string
	Phone     string
	CreatedAt time.Time
}

func (r Registration) CreatedAtFormatted() string {
	return r.CreatedAt.UTC().Format("2006-01-02 15:04:05") + " UTC"
}
